### PR TITLE
Camouflage timers will no longer be reset by PvE attacks. Grammar Correction

### PR DESCRIPTION
**Pull request by jaystewart1976**
Currently Camouflage looks like this

```
if (player.CurrentRegion.IsRvR && changeTime < DISABLE_DURATION)
          {
              player.Out.SendMessage(
                  LanguageMgr.GetTranslation(player.Client.Account.Language,
                      "Skill.Ability.CannotUse.Camouflage.DisableDuration", ((DISABLE_DURATION - changeTime) / 1000)),
                  eChatType.CT_System, eChatLoc.CL_SystemWindow);
              return;
          }
```

But
- Camouflage timers will no longer be reset by PvE attacks. Only an attack by the Archer on another player or a pet or turret controlled by another player will reset the timer. Camouflage is designed to let the Archer scout in PvP situations, so PvE interactions should not affect its timer.

http://camelotherald.wikia.com/wiki/Patch_Notes:_Version_1.62

So I have added another check 

```
if(!player.CurrentRegion.IsRvR)
          {
              new CamouflageEffect().Start(player);
                return;
            }
```

Which works as it should, I am not sure why this ability would be need in PVE but people spot a bug and it needs to be fixed.

```
CamouflageEffect camouflage = player.EffectList.GetOfType<CamouflageEffect>();
         if (camouflage != null)
         {
            camouflage.Cancel(false);
            return;
         }
         long changeTime = player.CurrentRegion.Time - player.LastAttackTickPvP;
          if (player.CurrentRegion.IsRvR && changeTime < DISABLE_DURATION)
          {
              player.Out.SendMessage(
                  LanguageMgr.GetTranslation(player.Client.Account.Language,
                      "Skill.Ability.CannotUse.Camouflage.DisableDuration", ((DISABLE_DURATION - changeTime) / 1000)),
                  eChatType.CT_System, eChatLoc.CL_SystemWindow);
              return;
          }
          if(!player.CurrentRegion.IsRvR)
          {
              new CamouflageEffect().Start(player);
                return;
            }
         player.DisableSkill(ab, DISABLE_DURATION);
         new CamouflageEffect().Start(player);
```

Jay.

### CODE REVIEW

**Comment by Graveen** [comment]
Ty for the GH Push !
Looking more closely, the important is DisableSkill which should not be fired in PvE :)

**Comment by jaystewart1976** [comment]
So if player zone !rvr disable skill altogether? I will change later

_Sent from my Samsung SM-G900F using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by OntheDOL** [comment]
Thanks for bringing this bug to attention! My only concern is the IsRvR is possibly not very PvP server friendly. (Im not at home so I cant view the code to see how isrvr is used around the place)

Perhaps a check to see if target is player/ pet player owner

**Comment by jaystewart1976** [comment]
Is it not possible to something to check server type like 
`if (player.CurrentRegion.IsRvR && changeTime < DISABLE_DURATION || GameServer.Instance.Configuration.ServerType == eGameServerType.GST_PvP)
{Use Timer}`

**Comment by Graveen** [comment]
Good point OnTheDOL. You can  use server rules to check this iirc

**Comment by OntheDOL** [comment]
hmm, now that I actually look at this, I'm not sure. From what i read, camouflage reuse timer should not be reset by attack in PvE. So if my reuse timer is say... 2:45, if i attack a monster, my timer remains unchanged. Now, if i attack a player, the timer is reset from 2:45 back to 7:00. From what i can tell , this is currently how it works. This looks like it would completely bypass the reuse timer if a player in not in an RvR zone. My impression on skill is: Attack mob, no affect on reuse timer, mob attack you, no affect on reuse timer, player attacks you, no affect on reuse timer, Attack player/pet/turrent, reuse timer is set back to full (ie 7:00)

**Comment by jaystewart1976** [comment]
This works 


`if (player.CurrentRegion.IsRvR && changeTime < DISABLE_DURATION)
            {
                player.Out.SendMessage(
                    LanguageMgr.GetTranslation(player.Client.Account.Language,
                        "Skill.Ability.CannotUse.Camouflage.DisableDuration", ((DISABLE_DURATION - changeTime) / 1000)),
                    eChatType.CT_System, eChatLoc.CL_SystemWindow);
                return;
            }
           
            if (GameServer.Instance.Configuration.ServerType == eGameServerType.GST_PvP && changeTime < DISABLE_DURATION)
            {
                player.Out.SendMessage(
                    LanguageMgr.GetTranslation(player.Client.Account.Language,
                        "Skill.Ability.CannotUse.Camouflage.DisableDuration", ((DISABLE_DURATION - changeTime) / 1000)),
                    eChatType.CT_System, eChatLoc.CL_SystemWindow);
                return;
            }

            if (!player.CurrentRegion.IsRvR && GameServer.Instance.Configuration.ServerType  != eGameServerType.GST_PvP)
            {
                
               new CamouflageEffect().Start(player);
                return;
            }`

**Comment by jaystewart1976** [comment]
[CamouflageSpecHandler.zip](https://github.com/Dawn-of-Light/DOLSharp/files/1072060/CamouflageSpecHandler.zip)

File: GameServer/skillhandler/CamouflageSpecHandler.cs [hunk 1]
```diff
@@ -23,64 +23,72 @@
...
+            long changeTime = player.CurrentRegion.Time - player.LastAttackTickPvP;
+            if (player.CurrentRegion.IsRvR && changeTime < DISABLE_DURATION)
+            {
+                player.Out.SendMessage(
+                    LanguageMgr.GetTranslation(player.Client.Account.Language,
+                        "Skill.Ability.CannotUse.Camouflage.DisableDuration", ((DISABLE_DURATION - changeTime) / 1000)),
+                    eChatType.CT_System, eChatLoc.CL_SystemWindow);
+                return;
+            }
+            if (!player.CurrentRegion.IsRvR)
```
Comment (dol-leodagan) [inline GameServer/skillhandler/CamouflageSpecHandler.cs]: Some Comment on this code : 

Why use an if clause to duplicate start effect then "return" and prevent DisableSkills from triggering, instead of just putting the DisableSkill trigger in an if clause itself ?

**Comment by dol-leodagan** [comment]
You're posting Code (in a zip !) as comment in GitHub ? This looks like a really inefficient way to use this website...

You known you can comment code part directly in the Diff display ?
You can also create a branch from any point of your code and try some edits there then throw a linked diff in the comment...

**Comment by jaystewart1976** [comment]
WoW.... Well I highlighted bug, tried a fix my bad. I will leave it to you guys.

File: GameServer/skillhandler/CamouflageSpecHandler.cs (hunk 1 above)
Comment (jaystewart1976) [inline GameServer/skillhandler/CamouflageSpecHandler.cs]: Please, I don't fully understand I don't want to disable the skill if it isn't an RvR Zone and Server type isn't PvP.
` if (!player.CurrentRegion.IsRvR && GameServer.Instance.Configuration.ServerType  != eGameServerType.GST_PvP)
        {
            
           new CamouflageEffect().Start(player);
            return;
        }``

**Comment by Graveen** [comment]
Using GH is good, and thank you for your 1st PR Jay.
We will both see the actual patch (i don't myself understand what is the correct behaviour) and the way to patch this.
Leodagan did general GH usage, nothing rude nor personal ;)

**Comment by OntheDOL** [comment]
GitHub is a great platform for purposing your code but also receiving criticism too, and everybody gets criticised, Graveen and Leodagan included ;) I don't believe anything personal is meant, we all want good code to enter the DOL source. I have learnt a lot from feedback from my PR's and I believe my coding style has improved from it. 

Now onto this PR, you are close, but i believe there are some changes that need to be made. I logged onto pendragon with two accounts to check this ability out , as i have little knowledge of it. A quick test showed the following: Using the ability, NO disableskill timer is set. Attacking anything pve, anywhere, RvR zone or normal, NO disableskil timer is set. 

The only time the skill was put into disableskill timer was when i attacked a player, with the skill on or off, it didnt matter, skill was put onto disable timer. Once the skill was disabled, the disable timer was reset upon attacking a player again, and it could not be enabled in any zone until timer was done. Timer was 10 minutes. A player attacking me did not set the skill into diabletimer mode.

So from this i think there should not be any disableskill calls in the handler at all. The disableskill call in the GamePlayer MakeAttack method should be slightly modified so if the skill is not currently disabled, it gets put on diableskill, if skill IS on disableskill timer, then do nothing. 

I will suggest a code change later when i get time.

TLDR: yes there is a bug with the skill, no this does not fix it ( In my opinion)

**Comment by jaystewart1976** [comment]
The ability to live test 👍 . Will I get to see the changes you have made as it will give me a much better insight into how it all links and help me going forward. Thank you guys.
